scipy/sparse/linalg/_interface.py

```diff
@@ -394,6 +391,12 @@ def __call__(self, x):
     def __mul__(self, x):
         return self.dot(x)
 
+    def __truediv__(self, other):
+        if not np.isscalar(other):
+            raise ValueError("Can only divide a linear operator by a scalar.")
```

Comment:
I discussed this PR in person briefly with @perimosocordiae, and he mentioned we might want to consider `NotImplementedError` for delegation reasons? Does that make sense?

Apart from that, he mostly seemed happy here.

Reply:
I have considered switching this to `return NotImplemented`, and after looking at the current test case I realized taht it is rather dangerous.

Consider what happens in the test that I have added for this error: there we have something like `aslinearoperator(np.eye(5)) / np.eye(5)` (a linear operator divided by an array). If I change this line to `return NotImplemented`, the following would happen:
1. `self` returns `NotImplemented` and delegates division to rhs.
2. Numpy tries to apply right division of the numerator by every element of the array.
3. This now works because `LinearOperator` now implements division
4. We get an object array of the same shape as the denominator, where every entry is a `_ScaledLinearOperator`. :zany_face: 

I conclude that delegating the division logic to the rhs would make a possible error result in an unexpected and uncaught data corruption. I therefore suggest to leave generalization of this code for having a more concrete use case in mind.

Does this sound reasonable?